timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py

```diff
@@ -131,24 +139,27 @@ def preprocess(
                 data[self.target] = data[self.target].fillna(value=self._train_target_median)
         return data, known_covariates
 
-    def _get_extra_tabular_init_kwargs(self) -> dict:
+    def _get_extra_tabular_init_kwargs(self) -> Dict[str, Any]:
         raise NotImplementedError
 
-    def get_hyperparameters(self) -> dict:
-        model_params = super().get_hyperparameters().copy()
-        model_params.setdefault("max_num_items", 20_000)
-        model_params.setdefault("max_num_samples", 1_000_000)
-        model_params.setdefault("tabular_hyperparameters", {"GBM": {}})
-        model_params.setdefault("tabular_fit_kwargs", {})
-        return model_params
+    def _get_default_hyperparameters(self) -> Dict[str, Any]:
+        return {
+            "max_num_items": 20_000,
+            "max_num_samples": 1_000_000,
+            "tabular_hyperparameters": {"GBM": {}},
+            "tabular_fit_kwargs": {},
+        }
 
-    def _get_mlforecast_init_args(self, train_data: TimeSeriesDataFrame, model_params: dict) -> dict:
+    def _get_mlforecast_init_args(
+        self, train_data: TimeSeriesDataFrame, model_params: Dict[str, Any]
+    ) -> Dict[str, Any]:
         from mlforecast.target_transforms import Differences
 
         from .transforms import MLForecastScaler
 
         lags = model_params.get("lags")
         if lags is None:
+            assert self.freq is not None
```

Comment:
Side note: This is something that we need to assert in `AbstractTimeSeriesModel` (or make `freq` a require argument to `AbstractTimeSeriesModel.__init__`). Most models implicitly assume that `self.freq` is available, e.g., to generate the forecast horizon index.

Reply:
added a TODO